examples/standalone/acoustic_comms_demo/acoustic_comms_demo.cc

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * Check README for detailed instructions.
+ * Usage:
+ *   $ gz sim -r worlds/acoustic_comms_demo.sdf
+ *   $ acoustic_comms_demo
+ */
+
+#include <atomic>
+#include <chrono>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include <gz/msgs.hh>
+#include <gz/transport.hh>
+
+using namespace gz;
+
+std::atomic<bool> tethysMsgReceived = false;
+std::atomic<bool> daphneMsgReceived = false;
+
+void cbTethys(const msgs::Dataframe &_msg)
```

Comment:
@adityapande-1995 nit: there is an `gz::transport::Node::Subscribe` overload that takes `std::function` objects. We could use lambdas instead of free functions and globals :eyes: :eyes: :eyes:

Reply:
Done https://github.com/gazebosim/gz-sim/pull/1693/commits/71cc56713cf704e559e22f31323c9995dfe800c9